### PR TITLE
ULMFiT AWD-QRNN

### PR DESCRIPTION
**Pull request by meghanabhange**

### CODE REVIEW

File: HinglishULMFiT_AWD_QRNN.ipynb [hunk 1]
```diff
@@ -0,0 +1,364 @@
+{
```
Comment (NirantK) [inline HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>What does <code>ninja</code> do?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NirantK/Hinglish/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (NirantK) [inline HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>Do we have another experiment like this with Stanza tokenizer instead of <code>spm</code>?</p><p> </p><p>Just asking out of curiosity.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NirantK/Hinglish/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (NirantK) [inline HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>What was the sp model trained on? Can we add that as a comment here please?</p><p> </p><p>We should add this somewhere in the repo README and paper as well maybe?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NirantK/Hinglish/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (NirantK) [inline HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>How are the params selected?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NirantK/Hinglish/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (NirantK) [inline HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>What is the <code>i</code>  here? Is this manually updated at every notebook run?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NirantK/Hinglish/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (NirantK) [inline HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>Here is a trick for generating names for models without using the <code>i</code>  mess which I'd used: </p><p> </p><p>Use something like <code>datetime.now().date()</code>.  Typically, we get one good model per day so this works. Is there are more, just use the hour in addition to date. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NirantK/Hinglish/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Required for qrnn :(

**Comment by NirantK** [comment]
@meghanabhange this branch with AWD-QRNN is still waiting updates from you. Please look into this? Or if you think this is not important - let's close this PR/del this branch without merging?

File: HinglishULMFiT_AWD_QRNN.ipynb (hunk 1 above)
Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Nope, we didn't try that.

Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Well, used this because sometimes the model at 7th epoch performed better than 9th. So using `i` was easier to select that model.

Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: <p>Used <code>data/interim/hinglish_unsup_high_confidence.json</code> and <code>data/interim/hinglish_unsup_less_confidence.json</code>.   The training notebook is <a href="https://github.com/NirantK/Hinglish/blob/sentencepiece/SentencePieceTokenizer.ipynb" target="_blank">here</a></p><p>Some details- </p><ul><li>Vocab - 8k </li><li>Training data size - 540k tweets</li><li>model_type - bpe</li></ul>

Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: <ul><li>Learning rate - Based on suggestion from <code>lr_find()</code></li><li>Epochs - randomly based on which ones gave better results. </li></ul><p> </p>

Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: https://github.com/NirantK/Hinglish/pull/31#discussion_r477562476

Comment (NirantK) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Use best model selection instead? Like why save every epoch? Just save top 2 best on validation set?

Comment (NirantK) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Add results of `lr_find` in the notebook somewhere?

Comment (NirantK) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Let's move these details to Markdown for easier accessibility and re-use for later experiments? Create a table/leaderboard which notes these somewhere? We are basically benchmarking stuff and if we don't document our config - we waste effort by either losing experiments or repeating them.

Comment (NirantK) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Cool, let's mark that for later then.

Comment (meghanabhange) [reply HinglishULMFiT_AWD_QRNN.ipynb:1]: Yeppp, should I add this in another PR that we will create for documentation?

**Comment by meghanabhange** [comment]
Let's close this PR because it will create more conflicts? 

- Will add metrics and info about sentencepiece in markdown with all the changes for documentation PR?
- Will create a separate PR for with changes in ULMFiT notebook that will include best-practices for saving models?

**Comment by meghanabhange** [comment]
Let's close this?

**Comment by NirantK** [comment]
Delete the branch if you'd like @meghanabhange
